### DISCUSSION TITLE
VaultUnlockEvent

### DISCUSSION THREAD

**Discussion by Grabsky** [PaperMC/Paper#11687] [Ideas]
### Is your feature request related to a problem?

Currently, it's quite tricky to listen to (and cancel) vault unlocking without listening to `PlayerInteractEvent`.

### Describe the solution you'd like.

`VaultUnlockEvent` which would be called when player (successfully) unlocks a vault with a key. This event should be cancellable.

### Describe alternatives you've considered.

`PlayerInteractEvent` is what I'm using currently. While it works, it isn't as reliable as `VaultUnlockEvent` could be, mainly because logic is based on predictions of whether vault actually unlocked or not.

**Comment by Warriorrrr** [comment]
The [BlockDispenseLootEvent](https://jd.papermc.io/paper/1.21.4/org/bukkit/event/block/BlockDispenseLootEvent.html) seems to already exist for this case now, a new event won't be able to expose anything extra (at least once vault api is merged in)

**Reply by Grabsky** [reply]
Would cancelling BlockDispenseLootEvent prevent consumption of the configured key? Name does not suggest that but I'm unable to check this right now. Generally this is the motivation behind my suggestion.
